src/serve/database.ts

```diff
@@ -324,9 +324,9 @@ export const initDB = async ({ skipDbPreloading }: { skipDbPreloading?: boolean
       .filter(k => {
         if (k.length !== HASH_LENGTH) return false
         const parsed = maybeParseCID(k)
-        return ([
+        return parsed && ([
           multicodes.SHELTER_CONTRACT_MANIFEST,
-          multicodes.SHELTER_CONTRACT_TEXT].includes(parsed?.code ?? -1)
+          multicodes.SHELTER_CONTRACT_TEXT].includes(parsed.code)
```

Comment:
`?? -1` this was just a TS band-aid.

Getting this if reverting to legacy code:
```
TS2345 [ERROR]: Argument of type 'number | undefined' is not assignable to parameter of type 'number'.
  Type 'undefined' is not assignable to type 'number'.
          multicodes.SHELTER_CONTRACT_TEXT].includes(parsed?.code) 
 ```
 
 This fix changes the logic to:
1. First check if `parsed` exists (`parsed &&`)
2. Only then access `parsed.code` (without the optional chaining since we know parsed is truthy)

This matches the pattern used elsewhere in the codebase, like in `routes.ts` where we see:

```.ts
const parsed = maybeParseCID(hash)
if (!parsed) {
  return Boom.badRequest()
}
```